### PR TITLE
feat(query): migrate date to number functions to v2.

### PR DESCRIPTION
**Pull request by RinChanNOWWW**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- to_yyyymm
- to_yyyymmdd
- to_yyyymmddhhmmss
- to_year
- to_month
- to_day_of_year
- to_day_of_month
- to_day_of_week
- to_hour
- to_minute
- to_second

Part of #8048

### CODE REVIEW

File: src/query/expression/src/date_helper.rs [hunk 1]
```diff
@@ -161,3 +162,104 @@ impl AddTimesImpl {
...
+    fn to_number(dt: &DateTime<Tz>) -> u32 {
+        dt.year() as u32 * 10000 + dt.month() * 100 + dt.day()
+    }
+}
+
+impl ToNumber<u64> for ToYYYYMMDDHHMMSS {
+    fn to_number(dt: &DateTime<Tz>) -> u64 {
+        dt.year() as u64 * 10000000000
+            + dt.month() as u64 * 100000000
+            + dt.day() as u64 * 1000000
```
Comment (sundy-li) [inline src/query/expression/src/date_helper.rs:212]: Use commas to split long numbers.   `1,000,000`

**Review by sundy-li — APPROVED** [review]

File: src/query/expression/src/date_helper.rs (hunk 1 above)
Comment (Xuanwo) [reply src/query/expression/src/date_helper.rs:212]: maybe `1_000_000`?

File: Cargo.lock [hunk 2]
```diff
@@ -4662,12 +4662,6 @@ dependencies = [
  "digest 0.10.5",
 ]
 
-[[package]]
```
Comment (Xuanwo) [inline Cargo.lock:4665]: Merge with main branch to resolve this.

**Comment by Xuanwo** [comment]
@mergify update

**Pull request merged**